src/collection.ts

```diff
@@ -53,9 +59,358 @@ export abstract class ReadOnlyCollection<T extends TigrisCollectionType> impleme
+	 * @returns {@link UpdateResponse}
+	 *
+	 * @example To update **language** of a book published by "Marcel Proust"
+	 * ```
+	 * const updatePromise = db.getCollection<Book>(Book).updateOne({
+	 *   filter: { author: "Marcel Proust" },
+	 *   fields: { language: "French" }
+	 * });
+	 *
+	 * updatePromise
+	 * 		.then((resp: UpdateResponse) => console.log(resp));
+	 * 		.catch( // catch the error)
+	 * 		.finally( // finally do something);
+	 * ```
+	 */
+	updateOne(query: UpdateQuery<T>): Promise<UpdateResponse>;
+
+	/**
+	 * Update a single document in a collection in transactional context
+	 *
+	 * @param query - Filter to match the document and update operations. Update
+	 * 								will be applied to a single matching document only.
+	 * @param tx - Session information for transaction context
+	 * @returns {@link UpdateResponse}
+	 *
+	 * @example To update **language** of a book published by "Marcel Proust"
+	 * ```
+	 * const updatePromise = db.getCollection<Book>(Book).updateOne({
+	 *   filter: { author: "Marcel Proust" },
+	 *   fields: { language: "French" }
+	 * }, tx);
+	 *
+	 * updatePromise
+	 * 		.then((resp: UpdateResponse) => console.log(resp));
+	 * 		.catch( // catch the error)
+	 * 		.finally( // finally do something);
+	 * ```
+	 */
+	updateOne(query: UpdateQuery<T>, tx: Session): Promise<UpdateResponse>;
+
+	updateOne(query: UpdateQuery<T>, tx?: Session): Promise<UpdateResponse> {
+		if (query.options === undefined) {
+			query.options = new UpdateQueryOptions(1);
+		} else {
+			query.options.limit = 1;
+		}
+		return this.updateMany(query, tx);
+	}
+
+	/**
+	 * Delete documents from collection matching the query
+	 *
+	 * @param query - Filter to match documents and other deletion options
+	 * @returns {@link DeleteResponse}
+	 *
+	 * @example
+	 *
+	 * ```
+	 * const deletionPromise = db.getCollection<Book>(Book).deleteMany({
+	 * 		filter: { author: "Marcel Proust" }
+	 * });
+	 *
+	 * deletionPromise
+	 * 		.then((resp: DeleteResponse) => console.log(resp));
+	 * 		.catch( // catch the error)
+	 * 		.finally( // finally do something);
+	 * ```
+	 */
+	deleteMany(query: DeleteQuery<T>): Promise<DeleteResponse>;
+
+	/**
+	 * Delete documents from collection in transactional context
+	 *
+	 * @param query - Filter to match documents and other deletion options
+	 * @param tx - Session information for transaction context
+	 * @returns {@link DeleteResponse}
+	 *
+	 * @example
+	 *
+	 * ```
+	 * const deletionPromise = db.getCollection<Book>(Book).deleteMany({
+	 * 		filter: { author: "Marcel Proust" }
+	 * }, tx);
+	 *
+	 * deletionPromise
+	 * 		.then((resp: DeleteResponse) => console.log(resp));
+	 * 		.catch( // catch the error)
+	 * 		.finally( // finally do something);
+	 * ```
+	 */
+	deleteMany(query: DeleteQuery<T>, tx: Session): Promise<DeleteResponse>;
+
+	deleteMany(query: DeleteQuery<T>, tx?: Session): Promise<DeleteResponse> {
+		return new Promise<DeleteResponse>((resolve, reject) => {
```

Comment:
Very minor. But with a few of these functions, you could move the `Promise` to only wrap the grpc call. That way you don't need to call reject everywhere and instead do a `throw`. I'm not really sure if it would make the code any clearer. But I wanted to mention it just in case.